src/clinical/service-worker-thread/tasks.ts

```diff
@@ -58,21 +60,21 @@ const isEntityInQuery = (entityName: ClinicalEntitySchemaNames, entityTypes: str
 // Main Sort Function
 const sortDocs = (
 	sortQuery: string,
-	entityName: string,
 	completionStats: CompletionDisplayRecord[],
+	sortType: keyof typeof ClinicalDataSortTypes,
```

Comment:
So this second point is worth discussing in depth. I don't think it's well defined what behaviour we want.

I approached this as, invalid records should be prioritized first by default until another sorting option is selected. You're seeing it as, invalid records should be prioritized first in every scenario.

My thinking is, if invalid records are always prioritized, and a User selects a specific Column to sort by, they aren't necessarily seeing the 'last' or 'highest' value at the top of that column. It disrupts the logical 'A-Z, 1-10' you'd expect to some degree. Maybe that's fine though, because error records will be sorted.


I solved this in such a way that invalid records are prioritized on first page load. They are also sorted by DonorId, so there is a combination.
But if a User chooses a specific sorting column, there's no way to go back to 'invalid record sorting' other than a page refresh.

Either way I don't think it's defined which behaviour is expected. Both have tradeoffs.